modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/criteria_builder/CriteriaRow.es.js

```diff
@@ -146,6 +150,61 @@ class CriteriaRow extends Component {
 		supportedPropertyTypes: {}
 	};
 
+	constructor(props, context) {
```

Comment:
I think grabbing context from the constructor is legacy https://reactjs.org/docs/legacy-context.html, here is the new one https://reactjs.org/docs/context.html#contextconsumer. Not an issue if it works, but it might stop working.

Reply:
Nice catch! 🎣Ya it'll be a good idea to update this, so I'll make the changes.

Reply:
Actually this was already refactored to call a method `_fetchEntityName` which uses `this.context`. I suppose that's another work around?

Reply:
Nicely done!!